creator-node/src/apiHelpers.js

```diff
@@ -191,3 +191,73 @@ module.exports.errorResponseNotFound = (message) => {
 module.exports.errorResponseSocketTimeout = (socketTimeout) => {
   return errorResponse(500, `${socketTimeout} socket timeout exceeded for request`)
 }
+
+/**
+ * Define custom api error subclasses to be thrown in components and handled in route controllers
+ */
+
+class ErrorBadRequest extends Error {}
+Object.defineProperty(ErrorBadRequest.prototype, 'name', {
+  value: 'ErrorBadRequest'
+})
+class ErrorServerError extends Error {}
+Object.defineProperty(ErrorServerError.prototype, 'name', {
+  value: 'ErrorServerError'
+})
+
+module.exports.ErrorBadRequest = ErrorBadRequest
+module.exports.ErrorServerError = ErrorServerError
+
+/**
+ * Given an error instance, returns the corresponding error response to request
+ * @param {Error} error instance of error class or subclass
+ */
+module.exports.handleApiError = (error) => {
+  // logger.info(`e: ${e} // ${e.name} // ${e.message}`)
+  switch (error) {
+    case ErrorBadRequest:
+      return this.errorResponseBadRequest(error.message)
+    case ErrorServerError:
+      return this.errorResponseServerError(error.message)
+    default:
+      return this.errorResponseServerError(error.message)
+  }
+}
+
+/**
+ * Helper function to parse responses from axios requests to other Content Nodes.
+ *    Given a response object and required fields, errors if any required fields missing.
+ *    Also errors if any signature fields missing.
+ *    Unnests response data.data and returns formatted data, along with raw response object.
+ *    Uses response schema defined above in successResponse()
+ * @param {Object} respObj original response object from axios request to content node
+ * @param {string[]} requiredFields
+ */
+module.exports.parseCNodeResponse = (respObj, requiredFields = []) => {
+  if (!respObj.data || !respObj.data.data) {
```

Comment:
we should have an condition here to check respObj like 
```
if (respObj && (!respObject.data || !respObj.data.data)) {
```

This could error if respObj is undefined

Reply:
don't think that really fixes problem? i figured it was ok bc either way this will error and the route will 500. eg what happens if respObj is null, then it hits below code and errors anyway. if anything i'd rather put `if (!respObj || !respObject.data || !respObj.data.data)`

wdyt?

Reply:
yeah that if check you laid out makes sense. i guess it throws either way anyway so may not matter?

Reply:
that was my thinking basically, it's gonna throw so why bother